docker/Dockerfile

```diff
@@ -30,28 +30,27 @@ ENV PUDL_INPUT=${CONTAINER_PUDL_WORKSPACE}/input
 ENV PUDL_OUTPUT=${CONTAINER_PUDL_WORKSPACE}/output
 ENV DAGSTER_HOME=${CONTAINER_PUDL_WORKSPACE}/dagster_home
 
-RUN mkdir -p ${PUDL_INPUT} ${PUDL_OUTPUT} ${DAGSTER_HOME}
+RUN mkdir -p ${PUDL_INPUT} ${PUDL_OUTPUT} ${DAGSTER_HOME} ${PUDL_REPO}
 
 # Copy dagster configuration file
 COPY docker/dagster.yaml ${DAGSTER_HOME}/dagster.yaml
 
+# Copy the cloned pudl repository into the container
+# This includes the .git directory, so it is a whole repo
+COPY --chown=${MAMBA_USER}:${MAMBA_USER} . ${PUDL_REPO}
+
```

Comment:
Copy the whole cloned repo into its own directory within the container.

Reply:
I'm not sure I understand.

You mean that by copying in the single file, and then building the container, we get a cached layer in the docker image that corresponds to the python environment having been installed (so long as the `conda-lock.yml` didn't change since last time we did it) but that precedes the copying of the rest of the  files in the repository over (which will change any time anything changes in the repo)?

Reply:
Okay, I've reinstated this ordering that allows the conda environment to be cached, and added a comment explaining why it's separate.